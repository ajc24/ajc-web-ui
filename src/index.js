/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
