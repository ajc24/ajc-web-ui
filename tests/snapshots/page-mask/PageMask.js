import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { PageMask } from '../../../src';

describe('PageMask', () => {
  describe('Default props and rendering - Non visible page mask', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageMask id="test-id" onClose={() => {}}>
            Test Page Mask text.
          </PageMask>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with grey theme', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageMask id="test-id" isDisplayed={true} colour="grey" onClose={() => {}}>
            Test Page Mask text.
          </PageMask>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Component with red theme', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageMask id="test-id" isDisplayed={true} colour="red" onClose={() => {}}>
            Test Page Mask text.
          </PageMask>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - onClose functionality not set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <PageMask id="test-id" isDisplayed={true}>
            Test Page Mask text.
          </PageMask>
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
