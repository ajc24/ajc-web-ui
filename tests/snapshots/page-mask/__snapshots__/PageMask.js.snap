// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageMask Default props and rendering - Non visible page mask verifies the snapshot for the component 1`] = `null`;

exports[`PageMask Transferred props and rendering - Component with grey theme verifies the snapshot for the component 1`] = `
Array [
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-page-mask ajc-width-4k"
    id="page-mask-test-id-root"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-center ajc-width-hd"
      >
        <div
          className="ajc-flex-column ajc-flex-justify-start ajc-spacing-top-large ajc-spacing-bottom-large ajc-width-hd-with-padding ajc-text-white ajc-font-default"
          id="page-mask-test-id-content"
        >
          Test Page Mask text.
        </div>
      </div>
    </div>
  </div>,
  <button
    className="ajc-flex-row ajc-button ajc-button-page-mask-close ajc-text-white ajc-font-default ajc-button-grey"
    id="page-mask-test-id-close-button"
    onClick={[Function]}
    title="Click to close"
  >
     X 
  </button>,
]
`;

exports[`PageMask Transferred props and rendering - Component with red theme verifies the snapshot for the component 1`] = `
Array [
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-page-mask ajc-width-4k"
    id="page-mask-test-id-root"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
    >
      <div
        className="ajc-flex-row ajc-flex-justify-center ajc-width-hd"
      >
        <div
          className="ajc-flex-column ajc-flex-justify-start ajc-spacing-top-large ajc-spacing-bottom-large ajc-width-hd-with-padding ajc-text-white ajc-font-default"
          id="page-mask-test-id-content"
        >
          Test Page Mask text.
        </div>
      </div>
    </div>
  </div>,
  <button
    className="ajc-flex-row ajc-button ajc-button-page-mask-close ajc-text-white ajc-font-default ajc-button-red"
    id="page-mask-test-id-close-button"
    onClick={[Function]}
    title="Click to close"
  >
     X 
  </button>,
]
`;

exports[`PageMask Transferred props and rendering - onClose functionality not set verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-row ajc-flex-justify-center ajc-page-mask ajc-width-4k"
  id="page-mask-test-id-root"
>
  <div
    className="ajc-flex-row ajc-flex-justify-center ajc-width-full-hd"
  >
    <div
      className="ajc-flex-row ajc-flex-justify-center ajc-width-hd"
    >
      <div
        className="ajc-flex-column ajc-flex-justify-start ajc-spacing-top-large ajc-spacing-bottom-large ajc-width-hd-with-padding ajc-text-white ajc-font-default"
        id="page-mask-test-id-content"
      >
        Test Page Mask text.
      </div>
    </div>
  </div>
</div>
`;
